tailormap-components/projects/core/src/lib/user-interface/attributelist/attributelist-filter-values-form/attributelist-filter-values-form.component.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  Component,
+  Inject,
+  OnInit,
+} from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { UniqueValues } from '../attributelist-common/attributelist-filter-models';
+
+@Component({
+  selector: 'tailormap-attributelist-filter-values-form',
+  templateUrl: './attributelist-filter-values-form.component.html',
+  styleUrls: ['./attributelist-filter-values-form.component.css'],
+})
+export class AttributelistFilterValuesFormComponent implements OnInit {
+
+  public values: UniqueValues;
+
+  public colName: string;
+
+  public allOn: boolean
+
+  constructor(private dialogRef: MatDialogRef<AttributelistFilterValuesFormComponent>,
+              @Inject(MAT_DIALOG_DATA) public data: any) {
+    this.colName = data.colName;
+    this.values = data.values;
+    this.allOn = true
+    this.updateSelected();
+  }
+
+  public ngOnInit(): void {
+  }
+
+  public updateSelected() {
+    // this.values lijkt geen array, om foutmelding in InetelliJ te voorkomen: isArray
+    if (Array.isArray(this.values)) {
+      this.allOn = this.values != null && this.values.every(v => v.select)
+    }
+  }
+
+  public someSelected() {
+    if (this.values === null) {
+      return false;
+    }
+    let result = false;
+    // this.values lijkt geen array, om foutmelding in InetelliJ te voorkomen: isArray
+    if (Array.isArray(this.values)) {
```

Comment:
this.values is ook geen array: het is een UniqueValues. Hierin zit wel een array, maar ook een boolean. Ik denk dat de "foutmelding" van intellij dus wel klopt

Reply:
UniqueValues bevat volgens mij een array van string en boolean, maar dat wordt blijkbaar niet herkend.
Op deze manier werkt het wel, maar mogelijk moet het nog net iets anders gecodeerd of gedeclareerd worden.

Reply:
wat je waarschijnlijk wilt doen is this.values.values (dat is de array). Hierop kan je je bewerkingen doen

Reply:
Dat had ik idd ook al geprobeerd, IntelliJ vind dat prachtig :-), maar er is uiteindelijk geen values in values. Als ik debug is this.values toch een array.
Ik heb het opgelost door de functies forEach, every en filter in het interface te declareren.
Dan is de fout weg en het werkt net als daarvoor.

Reply:
Uiteindelijk er achter gekomen dat ik UniqueValues niet hoef te gebruiken. Ook in *ngFor had ik vergelijkbaar probleem alhoewel het wel werkten.
Declareren van values: FilterValueSettings[] was de oplossing.